### PR TITLE
Adjust free license user checks

### PR DESCRIPTION
**Pull request by pjlast**
When adding the Free Plan instead of checking for nil, I missed a few spots.

## Test plan

Added unit tests

### CODE REVIEW

File: cmd/frontend/graphqlbackend/product_subscription_status.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func (productSubscriptionStatus) ProductNameWithBrand() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	hasLicense := info != nil
+	hasLicense := info != nil && !isFreePlan(info)
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/product_subscription_status.go:39]: Wouldn't it be better if `info` for free plan returns an empty `Tags()` slice? Or does that not work?

I'd love to see "free" become a proper [null object](https://golangbyexample.com/null-object-design-pattern-golang/)

File: cmd/frontend/graphqlbackend/product_subscription_status.go [hunk 2]
```diff
@@ -75,3 +75,13 @@ func (productSubscriptionStatus) MaximumAllowedUserCount(ctx context.Context) (*
 func (r productSubscriptionStatus) License() (*ProductLicenseInfo, error) {
 	return GetConfiguredProductLicenseInfo()
 }
+
+func isFreePlan(info *ProductLicenseInfo) bool {
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/product_subscription_status.go:79]: Suggestion: how about making this a method in `ProductLicenseInfo`?

File: cmd/frontend/graphqlbackend/product_subscription_status.go [hunk 3]
```diff
@@ -75,3 +75,13 @@ func (productSubscriptionStatus) MaximumAllowedUserCount(ctx context.Context) (*
...
+	for _, tag := range info.TagsValue {
+		if tag == "plan:free-0" {
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/product_subscription_status.go:81]:
```suggestion
		if tag == fmt.Sprintf("plan:%s", licensing.PlanFree0) {
```

I'm scared of this string being missed 😬

File: cmd/frontend/graphqlbackend/product_subscription_status_test.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+				TagsValue:      []string{"plan:business-0"},
+				UserCountValue: expectedUsers,
+				ExpiresAtValue: time.Now().Add(time.Hour * 8600),
+			}, nil
+		}
+		users, err := subscriptionStatus.MaximumAllowedUserCount(context.Background())
+		assert.NoError(t, err)
+		assert.Equal(t, int32(expectedUsers), *users)
+	})
+}
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/product_subscription_status_test.go:44]: Thanks for adding the tests! 💪 [Boy scout spirit, baby](https://deviq.com/principles/boy-scout-rule)

**Review by mrnugget — APPROVED** [review]

File: cmd/frontend/graphqlbackend/product_subscription_status.go (hunk 3 above)
Comment (pjlast) [reply cmd/frontend/graphqlbackend/product_subscription_status.go:81]: Would love to do that but `licensing` is an `internal` module. Can maybe do some other form of shuffling

**Pull request merged**